test/jasmine/tests/titles_test.js

```diff
@@ -0,0 +1,103 @@
+var d3 = require('d3');
+
+var Plotly = require('@lib/index');
+var interactConstants = require('@src/constants/interactions');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+var mouseEvent = require('../assets/mouse_event');
+
+describe('editable titles', function() {
+    'use strict';
+
+    var data = [{x: [1, 2, 3], y: [1, 2, 3]}];
+
+    var gd;
+
+    afterEach(destroyGraphDiv);
+
+    beforeEach(function() {
+        gd = createGraphDiv();
+    });
+
+    function checkTitle(letter, text, opacityOut, opacityIn) {
+        var titleEl = d3.select('.' + letter + 'title');
+        expect(titleEl.text()).toBe(text);
+        expect(+titleEl.style('opacity')).toBe(opacityOut);
+
+        var bb = titleEl.node().getBoundingClientRect(),
+            xCenter = (bb.left + bb.right) / 2,
+            yCenter = (bb.top + bb.bottom) / 2,
+            done,
+            promise = new Promise(function(resolve) { done = resolve; });
+
+        mouseEvent('mouseover', xCenter, yCenter);
+        setTimeout(function() {
+            expect(+titleEl.style('opacity')).toBe(opacityIn);
+
+            mouseEvent('mouseout', xCenter, yCenter);
+            setTimeout(function() {
+                expect(+titleEl.style('opacity')).toBe(opacityOut);
+                done();
+            }, interactConstants.HIDE_PLACEHOLDER + 50);
+        }, interactConstants.SHOW_PLACEHOLDER + 50);
+
+        return promise;
```

Comment:
Nicely done. :tada: 

If you're up for a challenge, you could try adding a test for title edits like I did [here](https://github.com/plotly/plotly.js/pull/1349/files#diff-7deb9e75965ca15196b29a1f7a4a9210R971) for legend/trace name edits.